calyx-frontend/src/syntax.pest

```diff
@@ -26,7 +27,8 @@ num_lit = ${
       ~ ( "d" ~ decimal
         | "b" ~ binary
         | "x" ~ hex
-        | "o" ~ octal )
+        | "o" ~ octal
+        | "f" ~ float)
```

Comment:
One challenging thing about this is that we are hardcoding IEEE-754 encoding for floating point and might eventually want to support things like posits (something that @Mark1626 was working on) so we might want a different way to specify floating point numbers?

@sampsyo thoughts? One thing I was thinking is that the syntax is only available in a magical primitive:
```C
cf1 = ieee754_const(5.11231); // Error if exact representation is not possible
```

Reply:
@jiahanxie353 What do you think about this approach? Instead of allowing floating point constants, we define a primitive `ieee754_const` and only allow the parser to parse floating point values within as its parameters. In all other cases, we throw an error saying that floating point constants must be generated using this special cell.

This will require some trickery in the parser/well-formedness checker but hopefully not too complicated?

Reply:
@jiahanxie353 ping about this!